package/MDAnalysis/core/universe.py

```diff
@@ -1289,6 +1289,69 @@ def _fragdict(self):
                 fragdict[a.ix] = fraginfo(i, f)
 
         return fragdict
+    
+    @classmethod
+    def from_smiles(cls, smiles, sanitize=True, addHs=True, 
+                    generate_coordinates=True, numConfs=1, **kwargs):
+        """Create a Universe from a SMILES string with rdkit
+
+        Parameters
+        ----------
+        smiles : str
+            SMILES string
+
+        sanitize : bool (optional, default True)
+            Toggle the sanitization of the molecule
+
+        addHs : bool (optional, default True)
+            Add all necessary hydrogens to the molecule
+
+        generate_coordinates : bool (optional, default True)
+            Generate 3D coordinates using RDKit's `AllChem.EmbedMultipleConfs`
+            function. Requires adding hydrogens with the `addHs` parameter
+
+        numConfs : int (optional, default 1)
+            Number of frames to generate coordinates for. Ignored if
+            `generate_coordinates=False`
+
+        kwargs : dict
+            Parameters passed on Universe creation
+
+        Returns
+        -------
+        :class:`~MDAnalysis.core.Universe`
+
+        Example
+        -------
+        To create a Universe with 10 conformers of ethanol ::
+        >>> u = mda.Universe.from_smiles('CCO', numConfs=10)
+        >>> u
+        <Universe with 9 atoms>
+        >>> u.trajectory
+        <RDKitReader with 10 frames of 9 atoms>
+        """
+        try:
+            from rdkit import Chem
+            from rdkit.Chem import AllChem
+        except ImportError as e:
+            raise ImportError(
+                "Creating a Universe from a SMILES string requires RDKit but " 
+                "it does not appear to be installed: {0}".format(e))
+
+        mol = Chem.MolFromSmiles(smiles, sanitize=sanitize)
+        if mol is None:
+            raise SyntaxError('Error while parsing SMILES {0}'.format(smiles))
+        if addHs:
+            mol = Chem.AddHs(mol)
+        if generate_coordinates:
+            if not addHs:
+                raise ValueError("Generating coordinates requires adding "
+                "hydrogens with `addHs=True`")
+            assert (type(numConfs) is int) and (numConfs > 0), ("numConfs must"
+            " be a non-zero positive integer instead of {0}".format(numConfs))
+            AllChem.EmbedMultipleConfs(mol, numConfs)
```

Comment:
EmbedMultipleConfs has a few useful options that users might want to set manually, is there any way we could capture these from **kargs?

Reply:
Since there are so many arguments that can be passed to this function, I went for a separate dict of kwargs. i.e. if you want to fix the random seed:
`mda.Universe.from_smiles("CCO", rdkit_kwargs=dict(randomSeed=42))`